components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java

```diff
@@ -449,9 +456,15 @@ public void updateConsumerApplication(OAuthConsumerAppDTO consumerAppDTO) throws
             oauthappdo.setScopeValidators(filterScopeValidators(consumerAppDTO));
             oauthappdo.setRequestObjectSignatureValidationEnabled(consumerAppDTO
                     .isRequestObjectSignatureValidationEnabled());
-            oauthappdo.setIdTokenEncryptionEnabled(consumerAppDTO.isIdTokenEncryptionEnabled());
-            oauthappdo.setIdTokenEncryptionAlgorithm(consumerAppDTO.getIdTokenEncryptionAlgorithm());
-            oauthappdo.setIdTokenEncryptionMethod(consumerAppDTO.getIdTokenEncryptionMethod());
+            // Validate IdToken Encryption configurations.
+            if (consumerAppDTO.isIdTokenEncryptionEnabled()) {
+                oauthappdo.setIdTokenEncryptionEnabled(consumerAppDTO.isIdTokenEncryptionEnabled());
+                if (consumerAppDTO.isIdTokenEncryptionEnabled()) {
```

Comment:
Isn't this redundant?

Reply:
Fixed in https://github.com/wso2-extensions/identity-inbound-auth-oauth/commit/4adb2395c29dd91bf7b6d835c2bb878f96f3741e#diff-1e769d65ab3dba0fc2f6ab9b801f1e72